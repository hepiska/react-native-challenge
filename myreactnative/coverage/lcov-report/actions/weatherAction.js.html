<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for actions/weatherAction.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">actions</a> weatherAction.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">42.86% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/4</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import axios from 'axios';
import { FETCH_WEATHER } from './constans';
&nbsp;
export const fetchWeatherDone = weathers =&gt; ({ type: FETCH_WEATHER, payload: weathers });
&nbsp;
export const fetchWeather = <span class="fstat-no" title="function not covered" >ci</span>ty =&gt; <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >di</span>spatch =&gt; (<span class="cstat-no" title="statement not covered" >axios.get('https://api.apixu.com/v1/forecast.json?key=146af849a55b4551877173858170205&amp;q=' + city + '&amp;days=2', {})</span></span>
.then(<span class="fstat-no" title="function not covered" >re</span>sponse =&gt; (<span class="cstat-no" title="statement not covered" >dispatch(fetchWeatherDone(response.data)))</span>)
.catch(<span class="fstat-no" title="function not covered" >er</span>r =&gt; <span class="cstat-no" title="statement not covered" >console.log(err))</span>
);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 01:33:02 GMT+0700 (WIB)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
