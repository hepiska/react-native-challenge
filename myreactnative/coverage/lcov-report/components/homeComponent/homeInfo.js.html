<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/homeComponent/homeInfo.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/homeComponent</a> homeInfo.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {
  View,
  Text,
  Image,
} from 'react-native';
import { connect } from 'react-redux';
import { Card, Grid, Col } from 'react-native-elements';
&nbsp;
const styles = {
  container: {
    width: '100%',
    height: '50%',
  },
  cardContainer: {
    width : '96%',
    height :'60%',
    marginTop:5,
    flexDirection: 'row',
  },
  cardColl: {
    width:"55%",
    margin: 2,
    height: '100%',
  }
};
&nbsp;
export const HomeInfo = props =&gt; {
  const { location, current } = props.weathers;
  if (!location) {
   return (&lt;Text&gt;Please Wait..&lt;/Text&gt;)
  }
  return (
    &lt;View style={styles.container}&gt;
      &lt;Card
        title={location.name}
        containerStyle={{ margin: 0 ,height:'100%'}}
        wrapperStyle={{ alignItems: 'center' }}
      &gt;
      &lt;Text&gt;{current.last_updated}&lt;/Text&gt;
      &lt;View style={styles.cardContainer}&gt;
        &lt;View style={styles.cardColl}&gt;
          &lt;Image source={{ uri: `https:${current.condition.icon}` }}
            style={{ width: '100%', height: '100%' }}
          /&gt;
        &lt;/View&gt;
        &lt;View&gt;
          &lt;Text&gt;Temp : {current.temp_c} celcius&lt;/Text&gt;
          &lt;Text&gt;Wind : {current.wind_kph} kph&lt;/Text&gt;
          &lt;Text&gt;Humidity : {current.humidity}&lt;/Text&gt;
        &lt;/View&gt;
      &lt;/View&gt;
      &lt;/Card&gt;
    &lt;/View&gt;
 )
}
const mapStateToprops = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ weathers: state.weathers })</span>;
export default connect(mapStateToprops, null)(HomeInfo);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 01:33:02 GMT+0700 (WIB)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
