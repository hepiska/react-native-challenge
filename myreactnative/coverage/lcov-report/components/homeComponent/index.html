<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/homeComponent</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> components/homeComponent
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.42% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/19</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.44% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">76.47% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/17</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="homeInfo.js"><a href="homeInfo.js.html">homeInfo.js</a></td>
	<td data-value="87.5" class="pic high"><div class="chart"><div class="cover-fill" style="width: 87%;"></div><div class="cover-empty" style="width:13%;"></div></div></td>
	<td data-value="87.5" class="pct high">87.5%</td>
	<td data-value="8" class="abs high">7/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file medium" data-value="searchBox.js"><a href="searchBox.js.html">searchBox.js</a></td>
	<td data-value="54.55" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 54%;"></div><div class="cover-empty" style="width:46%;"></div></div></td>
	<td data-value="54.55" class="pct medium">54.55%</td>
	<td data-value="11" class="abs medium">6/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 01:33:02 GMT+0700 (WIB)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
