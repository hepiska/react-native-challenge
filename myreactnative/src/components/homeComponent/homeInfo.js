import React from 'react';
import {
  View,
  Text,
  Image,
} from 'react-native';
import { connect } from 'react-redux';
import { Card, Grid, Col } from 'react-native-elements';

const styles = {
  container: {
    width: '100%',
    height: '50%',
  },
  cardContainer: {
    width : '96%',
    height :'60%',
    marginTop:5,
    flexDirection: 'row',
  },
  cardColl: {
    width:"55%",
    margin: 2,
    height: '100%',
  }
};

export const HomeInfo = props => {
  const { location, current } = props.weathers;
  if (!location) {
   return (<Text>Please Wait..</Text>)
  }
  return (
    <View style={styles.container}>
      <Card
        title={location.name}
        containerStyle={{ margin: 0 ,height:'100%'}}
        wrapperStyle={{ alignItems: 'center' }}
      >
      <Text>{current.last_updated}</Text>
      <View style={styles.cardContainer}>
        <View style={styles.cardColl}>
          <Image source={{ uri: `https:${current.condition.icon}` }}
            style={{ width: '100%', height: '100%' }}
          />
        </View>
        <View>
          <Text>Temp : {current.temp_c} celcius</Text>
          <Text>Wind : {current.wind_kph} kph</Text>
          <Text>Humidity : {current.humidity}</Text>
        </View>
      </View>
      </Card>
    </View>
 )
}
const mapStateToprops = state => ({ weathers: state.weathers });
export default connect(mapStateToprops, null)(HomeInfo);
