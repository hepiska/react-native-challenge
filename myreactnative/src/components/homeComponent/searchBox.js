import React from 'react';
import {
  View,
} from 'react-native';
import { SearchBar, Button } from 'react-native-elements';
import { connect } from 'react-redux';
import { fetchWeather } from '../../actions';

const styles = {
  container: {
    width: '100%',
    height: '16%',
  },
};

export class SearchBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = { text: '' };
  }
  search(text) {
    this.props.fetchWeathers(text);
    this.setState({ text: '' });
  }
  render() {
    return (
      <View style={styles.container}>
        <SearchBar
          lightTheme
          inputStyle={styles.input}
          value={this.state.text}
          onChangeText={text => this.setState({ text })}
          placeholder="Type Here..."
        />
        <Button
          small
          raised
          buttonStyle={{ marginLeft: 0, marginRight: 0 }}
          icon={{ name: 'search' }}
          title="Search"
          onPress={() => this.search(this.state.text)} />
      </View>
    );
  }
}
const mapDispatchToProps = dispatch => ({
  fetchWeathers: city => dispatch(fetchWeather(city)),
});
export default connect(null, mapDispatchToProps)(SearchBox);
