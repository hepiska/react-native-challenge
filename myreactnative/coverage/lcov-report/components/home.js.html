<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/home.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">components</a> home.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">55.56% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/6</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {
  Text,
  View,
} from 'react-native';
import { Button } from 'react-native-elements'
import { connect } from 'react-redux';
import {fetchWeather} from '../actions'
import { SearchBox,HomeInfo } from './homeComponent'
&nbsp;
export class Home extends React.Component {
  static navigationOptions = {
  title: 'Home',
  };
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    this.props.fetchWeathers('Jakarta');</span>
  }
  render() {
    const { navigate } = this.props.navigation;
    return (
      &lt;View style={{ flex: 1, alignItems: 'center', width:'100%' }} &gt;
        &lt;SearchBox /&gt;
        &lt;HomeInfo/&gt;
        &lt;View style={{width:'100%'}}&gt;
        &lt;Button
         icon={{name: 'cloud'}}
         title='Forecast'
         buttonStyle={{ marginLeft: 0, marginRight: 0 }}
         onPress={() =&gt; navigate('Weather')}
         /&gt;
        &lt;/View&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
const mapStateToprops = <span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ weathers: state.weathers})</span>;
const mapDispatchToProps = <span class="fstat-no" title="function not covered" >di</span>spatch =&gt; (<span class="cstat-no" title="statement not covered" >{ fetchWeathers: <span class="fstat-no" title="function not covered" >(c</span>ity) =&gt; <span class="cstat-no" title="statement not covered" >dispatch(fetchWeather(city)) </span>})</span>;
export default connect(mapStateToprops, mapDispatchToProps)(Home);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri May 05 2017 01:33:02 GMT+0700 (WIB)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
