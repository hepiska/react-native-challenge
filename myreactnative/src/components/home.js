import React from 'react';
import {
  Text,
  View,
} from 'react-native';
import { Button } from 'react-native-elements'
import { connect } from 'react-redux';
import {fetchWeather} from '../actions'
import { SearchBox,HomeInfo } from './homeComponent'

export class Home extends React.Component {
  static navigationOptions = {
  title: 'Home',
  };
  componentDidMount() {
    this.props.fetchWeathers('Jakarta');
  }
  render() {
    const { navigate } = this.props.navigation;
    return (
      <View style={{ flex: 1, alignItems: 'center', width:'100%' }} >
        <SearchBox />
        <HomeInfo/>
        <View style={{width:'100%'}}>
        <Button
         icon={{name: 'cloud'}}
         title='Forecast'
         buttonStyle={{ marginLeft: 0, marginRight: 0 }}
         onPress={() => navigate('Weather')}
         />
        </View>
      </View>
    );
  }
}

const mapStateToprops = state => ({ weathers: state.weathers});
const mapDispatchToProps = dispatch => ({ fetchWeathers: (city) => dispatch(fetchWeather(city)) });
export default connect(mapStateToprops, mapDispatchToProps)(Home);
