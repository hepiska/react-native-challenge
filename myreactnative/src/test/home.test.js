import React from 'react';
import { shallow, mount } from 'enzyme';
import ReactDOM from 'react-dom';
import {
  Text,
  View,
} from 'react-native';
import { Button } from 'react-native-elements';
import { connect } from 'react-redux';
import { fetchWeather } from '../actions';
import { SearchBox, HomeInfo } from '../components/homeComponent';
import { Home } from '../components/home';

describe ('home test', () => {
  const shallowrapper = shallow(<Home navigation={{ navigate: () => {} }} />);
  it('rendered', () => {
    expect(shallowrapper).toMatchSnapshot();
  });
  it('rendered', () => {
    shallowrapper.find(Button).simulate('press');
  });
});
