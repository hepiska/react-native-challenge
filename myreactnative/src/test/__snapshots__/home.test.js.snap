// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home test rendered 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "alignItems": "center",
            "flex": 1,
            "width": "100%",
          }
    }
>
    <Connect(SearchBox) />
    <Connect(HomeInfo) />
    <View
        style={
            Object {
                "width": "100%",
              }
        }
    >
        <Button
            buttonStyle={
                Object {
                    "marginLeft": 0,
                    "marginRight": 0,
                  }
            }
            icon={
                Object {
                    "name": "cloud",
                  }
            }
            onPress={[Function]}
            title="Forecast"
        />
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "alignItems": "center",
                  "flex": 1,
                  "width": "100%",
                }
      }
>
      <Connect(SearchBox) />
      <Connect(HomeInfo) />
      <View
            style={
                  Object {
                        "width": "100%",
                      }
            }
      >
            <Button
                  buttonStyle={
                        Object {
                              "marginLeft": 0,
                              "marginRight": 0,
                            }
                  }
                  icon={
                        Object {
                              "name": "cloud",
                            }
                  }
                  onPress={[Function]}
                  title="Forecast"
            />
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home
        navigation={
                Object {
                        "navigate": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "navigation": Object {
            "navigate": [Function],
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "alignItems": "center",
                              "flex": 1,
                              "width": "100%",
                            }
          }
>
          <Connect(SearchBox) />
          <Connect(HomeInfo) />
          <View
                    style={
                              Object {
                                        "width": "100%",
                                      }
                    }
          >
                    <Button
                              buttonStyle={
                                        Object {
                                                  "marginLeft": 0,
                                                  "marginRight": 0,
                                                }
                              }
                              icon={
                                        Object {
                                                  "name": "cloud",
                                                }
                              }
                              onPress={[Function]}
                              title="Forecast"
                    />
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "alignItems": "center",
                              "flex": 1,
                              "width": "100%",
                            }
          }
>
          <Connect(SearchBox) />
          <Connect(HomeInfo) />
          <View
                    style={
                              Object {
                                        "width": "100%",
                                      }
                    }
          >
                    <Button
                              buttonStyle={
                                        Object {
                                                  "marginLeft": 0,
                                                  "marginRight": 0,
                                                }
                              }
                              icon={
                                        Object {
                                                  "name": "cloud",
                                                }
                              }
                              onPress={[Function]}
                              title="Forecast"
                    />
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home
    navigation={
        Object {
            "navigate": [Function],
          }
    }
/>,
}
`;
